### ISSUE TITLE
Button `focus()` enhancements

### ISSUE THREAD

**Issue by ryanbmarx** [carbon-design-system/carbon-components-svelte#1830]
Currently, `Button.svelte` does not forward focus events. Adding `on:focus` to the element would solve this.

Also, to programmatically assign focus, access to the HTML element is required, but currently the Button component does not allow access. Perhaps this would work:

```
export let el: HTMLElement;

<button bind:this={el}
```

Which would allow `buttonInstance.el.focus()` in parent components.

I'm happy to generate a PR if these approaches are acceptable.

**Comment by metonym** [comment]
Note that the `Button` does support programmatic focus via the `ref` prop:

```html
<script>
  import { Button } from "carbon-components-svelte";

  let ref;

  $: ref?.focus();
</script>

<Button bind:ref>Primary button</Button>
```

However, I do think that forwarding `focus` to button events is reasonable. A PR would be welcome.

**Comment by metonym** [comment]
Fixed in [v0.82.8](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.82.8)
